src/azure-cli/azure/cli/command_modules/appconfig/_featuremodels.py

```diff
@@ -0,0 +1,275 @@
+                 name,
+                 parameters=None):
+        self.name = name
+        self.parameters = parameters
+
+    def __repr__(self):
+        featurefilter = {
+            "name": self.name,
+            "parameters": self.parameters
+        }
+        return json.dumps(featurefilter, indent=2)
+
+# Feature Flag Helper Functions #
+
+
+def custom_serialize_conditions(conditions_dict):
+    '''
+        Helper Function to serialize Conditions
+
+        Args:
+            conditions_dict - Dictionary of {str, List[FeatureFilter]}
+
+        Return:
+            JSON serializable Dictionary
+    '''
+    featurefilterdict = {}
+
+    for key, value in conditions_dict.items():
+        featurefilters = []
+        for featurefilter in value:
+            featurefilters.append(str(featurefilter))
+        featurefilterdict[key] = featurefilters
+    return featurefilterdict
+
+
+def map_keyvalue_to_featureflag(keyvalue, show_conditions=True):
+    '''
+        Helper Function to convert KeyValue object to FeatureFlag object for display
+
+        Args:
+            keyvalue - KeyValue object to be converted
+            show_conditions - Boolean for controlling whether we want to display "Conditions" or not
+
+        Return:
+            FeatureFlag object
+    '''
+    feature_name = keyvalue.key[len(FEATURE_FLAG_PREFIX):]
+
+    feature_flag_value = map_keyvalue_to_featureflagvalue(keyvalue)
+
+    state = FeatureState.OFF
+    if feature_flag_value.enabled:
+        state = FeatureState.ON
+
+    conditions = feature_flag_value.conditions
+
+    # if conditions["client_filters"] list is not empty, make state conditional
+    filters = conditions["client_filters"]
+
+    if filters and state == FeatureState.ON:
+        state = FeatureState.CONDITIONAL
+
+    feature_flag = FeatureFlag(feature_name,
+                               keyvalue.label,
+                               state,
+                               feature_flag_value.description,
+                               conditions,
+                               keyvalue.locked,
+                               keyvalue.last_modified)
+
+    # By Default, we will try to show conditions unless the user has
+    # specifically filtered them using --fields arg.
+    # But in some operations like 'Delete feature', we don't want
+    # to display all the conditions as a result of delete operation
+    if not show_conditions:
+        del feature_flag.conditions
+    return feature_flag
+
+
+def map_keyvalue_to_featureflagvalue(keyvalue):
+    '''
+        Helper Function to convert value string to a valid FeatureFlagValue.
+        Throws Exception if value is an invalid JSON.
+
+        Args:
+            keyvalue - KeyValue object
+
+        Return:
+            Valid FeatureFlagValue object
+    '''
+
+    default_conditions = {'client_filters': []}
+
+    try:
+        # Make sure value string is a valid json
+        feature_flag_dict = json.loads(keyvalue.value)
```

Comment:
CLI has [shell_safe_json_parse](https://github.com/Azure/azure-cli/blob/779ae8117ae8db7f0cc325932eeebeb683a34445/src/azure-cli-core/azure/cli/core/util.py#L238) for better json parse. If it is useful to me, you can reuse it.